### PR TITLE
[GLUTEN-5525][FOLLOWUP] Fix mvn versions:set does not work for shim submodules

### PR DESCRIPTION
**Pull request by JinHelin404**
## What changes were proposed in this pull request?

mvn versions:set does not work for pom.xml in the shim/spark3x directory.

## How was this patch tested?

Run commend like follow and check pom version.

```
mvn -N --batch-mode versions:set -DnewVersion=${spark_version} versions:update-child-modules
```

And cases mentioned in [5519](https://github.com/apache/incubator-gluten/pull/5519) are also tested.

```
mvn clean
mvn clean install -DskipTests
mvn clean -Pbackends-velox -DskipTests
```

### CODE REVIEW

**Comment by JinHelin404** [comment]
@PHILO-HE @zhztheplayer Could you please take a look? Thanks!

**Comment by PHILO-HE** [comment]
@JinHelin404, thanks for your pr!
Then, `sparkshim.module.name` is useless, can we remove it from property setting?

**Comment by JinHelin404** [comment]
@PHILO-HE Thanks for your reviewing! 

Sure we can. I removed the sparkshim.module.name property and added a default spark 34 profile in the pom of shims.

File: pom.xml [hunk 1]
```diff
@@ -309,7 +307,6 @@
       <properties>
         <sparkbundle.version>3.3</sparkbundle.version>
         <sparkshim.module.name>spark33</sparkshim.module.name>
-        <sparkshim.artifactId>spark-sql-columnar-shims-spark33</sparkshim.artifactId>
```
[the remark this answers is not in the document]
Comment (JinHelin404) [reply pom.xml:312]: Fixed.

**Comment by JinHelin404** [comment]
@PHILO-HE Could you please take a look again? Thank you!

**Review by PHILO-HE — APPROVED** [review]
Thanks!

**Pull request merged**